force-app/main/default/classes/ClassesObjectsMethods.cls

```diff
@@ -212,10 +216,13 @@ public with sharing class ClassesObjectsMethods {
      * @return The a positive number years between the two dates, or null if either date is null.
      */
     public static Integer calculateYearDifference(Date d1, Date d2) {
-
-        // Subtract the difference in years
-
-        return null; // Replace null with the variable you used to store the result
+        if(d1 != null && d2 != null) {
+            Integer d1Year = d1.year();
+            Integer d2Year = d2.year();
+            Integer differenceInYears = d2Year - d1Year;
```

Comment:
How can we always ensure the result is positive? What Math function can you use? This is important if the dates are reversed (eg, 1997 - 2024).